src/hotspot/share/classfile/systemDictionaryShared.cpp

```diff
@@ -1045,10 +1039,11 @@ InstanceKlass* SystemDictionaryShared::find_or_load_shared_class(
       // which are parallel-capable loaders, so this lock is NOT taken.
       Handle lockObject = compute_loader_lock_object(THREAD, class_loader);
       check_loader_lock_contention(THREAD, lockObject);
-      ObjectLocker ol(lockObject, THREAD, DoObjectLock);
+      ObjectLocker ol(lockObject, THREAD);
```

Comment:
I think we can replace the above 3 lines with something like
```
assert(!is_parallelCapable(class_loader), "ObjectLocker not required");
```

Reply:
Yes, I agree. I updated the comment slightly also.

Reply:
It's assert(is_parallelCapable(class_loader)).